apps/discovery_api/priv/static/tableau/connector.js

```diff
@@ -79,6 +79,11 @@ function _setupConnector() {
       .then(function(refreshToken) { tableau.password = refreshToken; })
       .catch(function(error) { _displayLoginError(error) })
       .then(function() { initCallback() })
+      .then(function() {
+        document.getElementById('login-text').textContent = 'Logged In';
+        document.getElementById('login-button').onclick = undefined;
+        document.getElementById('login-button').className = 'login-link';
+       })
```

Comment:
Should this go before the catch block?  
Also, how's about a test for this behavior?  😁